pkg/controller/staticroute/staticroute_controller.go

```diff
@@ -222,9 +223,12 @@ func reconcileImpl(params reconcileImplParams) (res *reconcile.Result, err error
 		}
 		// special error handling is needed in the following case
 		var serr error
-		if res == overlapsProtected {
+		switch res {
+		case overlapsProtected:
 			serr = errors.New("Given subnet overlaps with some protected subnet")
-		} else {
+		case gatewayNotDirectlyRoutableError:
+			serr = errors.New("Gateway IP is not directly routable, next hop detected")
```

Comment:
```suggestion
			serr = errors.New("The configured gateway IP is not directly routable, cannot setup the route.")
```
I would rephrase this error log to this.. I could do it. Thank you!